pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -254,8 +257,15 @@ private void setUpPinotController() {
 
     // Helix resource manager must be started in order to create PinotLLCRealtimeSegmentManager
     LOGGER.info("Starting realtime segment manager");
-    PinotLLCRealtimeSegmentManager
-        .create(_helixResourceManager, _config, _controllerMetrics, _controllerLeadershipManager);
+
+    _pinotLLCRealtimeSegmentManager =
+        new PinotLLCRealtimeSegmentManager(_helixResourceManager, _config, _controllerMetrics,
+            _controllerLeadershipManager);
+    // TODO: Need to put this inside HelixResourceManager when ControllerLeadershipManager is removed.
+    _helixResourceManager.registerPinotLLCRealtimeSegmentManager(_pinotLLCRealtimeSegmentManager);
```

Comment:
This line can be moved into the constructor of PinotLLCRealtimeSegmentManager?

Reply:
I've tried that but since the object isn't fully initialized, it will only assign a null reference to it.
Plus, the logic of these 2 objects are mixed together: 
* `Resource manager` has to be instantiated before `PinotLLCRealtimeSegmentManager`.
* `Resource manager` needs to call the method in `PinotLLCRealtimeSegmentManager`, which is created after `Resource manager` got created.

It's a circular reference dependency between these two objects.

Reply:
Make sense lol